fe/fe-core/src/main/java/com/starrocks/connector/iceberg/io/IcebergCachingFileIO.java

```diff
@@ -99,21 +104,48 @@ public class IcebergCachingFileIO implements FileIO {
     public static final long DISK_CACHE_EXPIRATION_SECONDS = Config.iceberg_metadata_disk_cache_expiration_seconds;
 
     private ContentCache fileContentCache;
-    private final FileIO wrappedIO;
+    private FileIO wrappedIO;
+    private Configuration conf;
+
+    public IcebergCachingFileIO() {
+    }
 
     public IcebergCachingFileIO(FileIO io) {
         this.wrappedIO = io;
     }
 
     @Override
     public void initialize(Map<String, String> properties) {
+        if (wrappedIO == null) {
+            switch (properties.get(ICEBERG_CATALOG_TYPE)) {
```

Comment:
What if using Rest Catalog, what `wrappedIO` should we use?

Reply:
Rest catalog doesn't need the file io field, the rest client will be created during initialization.